### PR TITLE
Use unified block production from updated eth2 client

### PR DESCRIPTION
**Pull request by nkryuchkov**
Since https://github.com/attestantio/go-eth2-client/pull/116 `GetBeaconBlock` can return both proposal and blinded proposal. 

This PR 
- removes the logic which eth2 client endpoint to choose 
- always uses `GetBeaconBlock` instead

TODO:

- tests

### CODE REVIEW

File: protocol/v2/ssv/spectest/msg_processing_type.go [hunk 1]
```diff
@@ -97,6 +97,14 @@ func (test *MsgProcessingSpecTest) compareBroadcastedBeaconMsgs(t *testing.T) {
...
 			}
 		}
+		if !found {
+			println("<<<<<<<<<<<<<<<<<<<<<<here>>>>>>>>>>>>>>>>>>>>>>")
+			println(r1)
+			for _, r2 := range broadcastedRoots {
+				println(hex.EncodeToString(r2[:]))
+			}
+			println("<<<<<<<<<<<<<<<<<<<<<<here>>>>>>>>>>>>>>>>>>>>>>")
+		}
```
Comment (moshe-blox) [inline protocol/v2/ssv/spectest/msg_processing_type.go:107]: revert?

File: protocol/v2/ssv/spectest/ssv_mapping_test.go [hunk 2]
```diff
@@ -43,7 +43,6 @@ func TestSSVMapping(t *testing.T) {
 	types.SetDefaultDomain(testingutils.TestingSSVDomainType)
 
 	for name, test := range untypedTests {
-		name, test := name, test
```
Comment (moshe-blox) [inline protocol/v2/ssv/spectest/ssv_mapping_test.go:46]: why delete?

**Review by moshe-blox — CHANGES_REQUESTED** [review]

**Review by moshe-blox — CHANGES_REQUESTED** [review]
just a few comments

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -40,7 +40,7 @@ Build stage Docker image:
     - docker tag $IMAGE_NAME:$CI_COMMIT_SHA $DOCKER_REPO_INFRA_STAGE:$CI_COMMIT_SHA
     - $DOCKER_LOGIN_TO_INFRA_STAGE_REPO && docker push $DOCKER_REPO_INFRA_STAGE:$CI_COMMIT_SHA
   only:
-    - stage
+    - use-updated-eth2-client
```
Comment (lior-blox) [inline .gitlab-ci.yml:43]: revert

File: scripts/differ/transformers.go [hunk 4]
```diff
@@ -8,6 +8,15 @@ import (
...
+func (t Transformers) Transform(code string) string {
+	for _, transformer := range t {
+		code = transformer(code)
+	}
+	return code
+}
+
 func NopTransformer() Transformer {
 	return func(code string) string { return code }
 }
```
Comment (y0sher) [inline scripts/differ/transformers.go:22]: imo bad practice this is here

File: e2e/go.mod [hunk 5]
```diff
@@ -3,11 +3,11 @@ module github.com/ssvlabs/ssv/e2e
 go 1.20
 
 require (
-	github.com/attestantio/go-eth2-client v0.19.11-0.20240129201044-9d799aaab2bd
+	github.com/attestantio/go-eth2-client v0.21.1-0.20240325145558-858fa8a893fa
```
Comment (y0sher) [inline e2e/go.mod:6]: what is this version? I thought we said lets update to [v0.21.3](https://github.com/attestantio/go-eth2-client/releases/tag/v0.21.3)
this commit is not an official tag - [858fa8a893fa](https://github.com/attestantio/go-eth2-client/commit/858fa8a893fa)

File: protocol/v2/ssv/spectest/ssv_mapping_test.go (hunk 2 above)
Comment (guym-blox) [reply protocol/v2/ssv/spectest/ssv_mapping_test.go]: double initialization of name,test

File: scripts/differ/transformers.go (hunk 4 above)
Comment (moshe-blox) [reply scripts/differ/transformers.go:22]: > imo bad practice this is here

why?

File: beacon/goclient/goclient.go [hunk 6]
```diff
@@ -182,6 +180,8 @@ func New(
 		// LogLevel supplies the level of logging to carry out.
 		eth2clienthttp.WithLogLevel(zerolog.DebugLevel),
 		eth2clienthttp.WithTimeout(commonTimeout),
+		eth2clienthttp.WithAllowDelayedStart(opt.DelayedStart),
```
[the remark this answers is not in the document]
Comment (moshe-blox) [reply beacon/goclient/goclient.go]: removed

**Review by y0sher — DISMISSED** [review]
would rather we use official spec tag but I think its best we don't wait for it.

**Comment by moshe-blox** [comment]
> would rather we use official spec tag but I think its best we don't wait for it.

yeah we need to start testing this

File: scripts/differ/transformers.go (hunk 4 above)
Comment (y0sher) [reply scripts/differ/transformers.go:22]: is this related to the change?

File: scripts/differ/main.go [hunk 7]
```diff
@@ -69,10 +70,10 @@ func run() (changes int, err error) {
 	}
 
 	// Prepare the transformers.
-	transformers := Transformers{
-		NoComments(),
-		NoPackageNames(config.ReducedPackageNames),
-		NoEmptyLines(),
+	transformers := transformer.Transformers{
```
Comment (y0sher) [inline scripts/differ/main.go:73]: sorry but this is not what I meant, I meant maybe this change shouldn't be part of this PR..

**Review by y0sher — APPROVED** [review]

**Review by moshe-blox — APPROVED** [review]
